drift/info_parser.py

```diff
@@ -255,69 +255,150 @@ def _create_comparison(systems, info_name, reference_id, system_count):
+                values["state"] = COMPARISON_SAME
+                if values["value"] != reference_value:
+                    values["state"] = COMPARISON_DIFFERENT
+
+        if (
+            len(sorted_system_id_values) - len(sorted_system_id_values_without_obfuscated)
+            > 0
+        ):  # when there is one or more obfuscated values
+            info_comparison = COMPARISON_INCOMPLETE_DATA_OBFUSCATED
+
+        return {
+            "name": info_name,
+            "state": info_comparison,
+            "systems": sorted_system_id_values,
+        }
 
-    return {
-        "name": info_name,
-        "state": info_comparison,
-        "systems": sorted_system_id_values,
-    }
+    else:
+        # multivalue fact handling
+        all_value_options = set()
+        sorted_system_values = []
+        if reference_id:
+            for system in sorted_system_id_values:
+                sorted_system_values.append(system["value"])
+                if reference_id == system["id"]:
+                    reference_value = system["value"]
+        else:
+            sorted_system_values = [system["value"] for system in sorted_system_id_values]
+
+        for value in sorted_system_values:
+            if not isinstance(value, list):
+                if value != "N/A":
+                    all_value_options.add(value)
+            else:
+                for item in value:
+                    all_value_options.add(item)
+
+        row_count = len(all_value_options)
+        sorted_all_value_options = sorted(list(all_value_options))
+        value_count = 0
+        multivalue_comparisons = [COMPARISON_SAME] * row_count
+        info_comparison = COMPARISON_SAME
+        for value in sorted_system_values:
+            row = 0
+            column = []
+            if not isinstance(value, list):
+                while row < row_count:
+                    if sorted_all_value_options[row] == value:
+                        column.append(value)
+                    else:
+                        column.append(None)
+                        info_comparison = COMPARISON_DIFFERENT
+                        multivalue_comparisons[row] = COMPARISON_DIFFERENT
+                    row += 1
+            else:
+                while row < row_count:
+                    if sorted_all_value_options[row] in value:
+                        column.append(sorted_all_value_options[row])
+                    else:
+                        column.append(None)
+                        info_comparison = COMPARISON_DIFFERENT
+                        multivalue_comparisons[row] = COMPARISON_DIFFERENT
+                    row += 1
+            sorted_system_id_values[value_count]["value"] = column
+            value_count += 1
+
+        # need to account for states compared to a reference if selected
+        #for system in sorted_system_id_values:
+        #    system["state"] = reference_comparison
+
+
+        row_count = len(multivalue_comparisons)
```

Comment:
This is redundant.  row_count value is set in line 350 and doesn't get changed after.